### DISCUSSION TITLE
html-fund-me-cu an error with `FundMe.sol` interaction

### DISCUSSION THREAD

**Discussion by DCVglobalnetwork** [Cyfrin/foundry-full-course-cu#2457] [General]
Hi 

I followed the instruction on the repo and with the course and this is the second time I am getting an error The first one worked but metamask interacted with the HTLM page without `FundMe.sol` contract interaction The first error was with HTML, which I do not remember, but I was able to interact between metamak and HTML page no. `FundMe.sol` 
Today I stumbed on another error below

![image](https://github.com/user-attachments/assets/2bbee6e0-432a-44f9-aa77-cdaaf12e5850)

im really confused 
Please help 

Thank you

**Comment by EngrPips** [comment]
Hello @DCVglobalnetwork, Please open your `lib` folder and show it.

**Reply by DCVglobalnetwork** [reply]
![image](https://github.com/user-attachments/assets/a19e4010-fdf4-4c93-b831-11ff794956aa)

![image](https://github.com/user-attachments/assets/d45ebd59-f7bd-4112-abfd-961d555879dd)

looks empty? but why I did clone the repo?

**Reply by EngrPips** [reply]
Run `forge build`, let us know what happens.

**Reply by DCVglobalnetwork** [reply]
![image](https://github.com/user-attachments/assets/a59b907c-c783-4877-800f-2bac8ff8e0e4)

**Reply by EngrPips** [reply]
Your stuff compiled successfully. So I think everything is good now.

**Reply by DCVglobalnetwork** [reply]
![image](https://github.com/user-attachments/assets/1cd9cf0b-0f54-49b6-80c6-34128b8de5aa)

should i modify the Makefile ? as is asking me for ETHEREUM API KEY

**Reply by DCVglobalnetwork** [reply]
I DID RUN THE ANVIL

**Reply by EngrPips** [reply]
You will need a `.env` file where you will provide all the values for the variables in your `Makefile`

**Reply by DCVglobalnetwork** [reply]
![image](https://github.com/user-attachments/assets/f1a0bac0-7644-4acc-b88a-18598539f48b)

Yes, I have. env but I thought I was not need it as i am using Anvil with what anvil provides

**Reply by EngrPips** [reply]
If you are using `Anvil` and things work fine then no worries.

**Reply by DCVglobalnetwork** [reply]
no they do not work that way because what Patrick is telling us to do is to clone repo. You have the previous one, `FundMe.sol` which he is telling us to clone repo HTML FUNDME to interact with Metamask and `FundMe.sol` using this HTML page, which than we must connect to wallet, so connected to wallet, go back to fundry-fundMe.sol to make anvil, and in new terminal make deply and this he is not using any API key from anywhere All is anvil with accounts and private keys that from foundry-fund-me he interacts with html-fund-me and with metamask I was able to interact only with HTML FundMe page and metamask but not as he does with fundry-fund-me, html fund me and metamask, so not i did not solve the error as i followed what he does, but I have an error Yes, i did deploy, but i was unable to interact with HTML page Fund Me, therefore I am a bit confused

**Reply by EngrPips** [reply]
I have forgotten what we did in that section. I will revisit and circle back here as soon as possible.

**Reply by EngrPips** [reply]
Hey @DCVglobalnetwork, what problem do you have while trying to interact with `HTML page Fund Me`? I just ran through the lesson now, and everything ran perfectly for me.

<img width="1440" alt="Screenshot 2024-09-12 at 22 00 11" src="https://github.com/user-attachments/assets/59ba9e7e-3fef-464a-a078-d84264e2ad83">
